code/modules/atmospherics/machinery/components/fusion/hfr_procs.dm

```diff
@@ -318,10 +318,100 @@
  * Create the explosion + the gas emission before deleting the machine core.
  */
 /obj/machinery/atmospherics/components/unary/hypertorus/core/proc/meltdown()
-	explosion(src, light_impact_range = power_level * 5, flash_range = power_level * 6, adminlog = TRUE, ignorecap = TRUE)
-	radiation_pulse(loc, power_level * 7000, (1 / (power_level + 5)), TRUE)
-	empulse(loc, power_level * 5, power_level * 7, TRUE)
-	var/list/around_turfs = circlerangeturfs(src, power_level * 5)
+	var/flash_explosion = 0
+	var/light_impact_explosion = 0
+	var/heavy_impact_explosion = 0
+	var/devastating_explosion = 0
+	var/emp_pulse = FALSE
+	var/rad_pulse = FALSE
+	var/emp_light_size = 0
+	var/emp_heavy_size = 0
+	var/rad_pulse_size = 0
+	var/rad_pulse_strenght = 0
+	var/gas_spread = 0
+	var/gas_pockets = 0
+	var/critical = FALSE
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_BASE_EXPLOSION)
+		flash_explosion = power_level * 3
+		light_impact_explosion = power_level * 2
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_MEDIUM_EXPLOSION)
+		flash_explosion = power_level * 6
+		light_impact_explosion = power_level * 5
+		heavy_impact_explosion = power_level * 0.5
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_DEVASTATING_EXPLOSION)
+		flash_explosion = power_level * 8
+		light_impact_explosion = power_level * 7
+		heavy_impact_explosion = power_level * 2
+		devastating_explosion = power_level
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_RADIATION_PULSE)
+		rad_pulse = TRUE
+	if(selected_fuel.meltdown_flags & HYPERTORUS_FLAG_EMP)
+		emp_pulse = TRUE
```

Comment:
To cut down on `if` spam, just assign these where the are declared to the result of the if expression

Reply:
what you mean by this?
should i do something like `emp_pulse = selected_fuel.meltdown_flags & HYPERTORUS_FLAG_EMP` ?